crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1180,6 +1181,252 @@ impl<'w> EntityWorldMut<'w> {
+        let to_delete = contributed_components
+            .iter()
+            .filter(|c| !remaining_bundle_info.contributed_components().contains(c))
+            .copied()
+            .collect::<Vec<_>>();
+
+        let to_delete_bundle = self.world.bundles.init_dynamic_info(components, &to_delete);
+        // SAFETY: the dynamic `BundleInfo` is initialized above
+        self.location = unsafe { self.remove_bundle(to_delete_bundle) };
+
+        self
+    }
+
+    /// Removes the components of this [`Bundle`], components that require them, and their required components,
+    /// unless they are required by components outside of this removal set.
+    ///
+    /// This method performs a top-down removal, starting by collecting all bundle components, all required components for bundle components, all components that require bundle components.
+    /// Then removing all collected components, unless those components are required outside remove set
+    /// elsewhere in the entity.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use bevy_ecs::prelude::*;
+    ///
+    /// #[derive(Component)]
+    /// #[require(B)]
+    /// struct A;
+    ///
+    /// #[derive(Component, Default)]
+    /// #[require(C)]
+    /// struct B;
+    ///
+    /// #[derive(Component, Default)]
+    /// #[require(D)]
+    /// struct C;
+    ///
+    /// #[derive(Component, Default)]
+    /// struct D;
+    ///
+    /// #[derive(Component)]
+    /// #[require(D)]
+    /// struct E;
+    ///
+    /// let mut world = World::new();
+    ///
+    /// // Spawn an entity with A, B, C, D, and E components
+    /// let entity = world.spawn((A, B, C, D, E)).id();
+    ///
+    /// // Initial component structure:
+    /// //  A -> B -> C -> D <- E
+    ///
+    /// assert!(world.entity(entity).contains::<A>());
+    /// assert!(world.entity(entity).contains::<B>());
+    /// assert!(world.entity(entity).contains::<C>());
+    /// assert!(world.entity(entity).contains::<D>());
+    /// assert!(world.entity(entity).contains::<E>());
+    ///
+    /// // Remove B and its requirement tree
+    /// world.entity_mut(entity).require_descendant_remove::<B>();
+    ///
+    /// // Resulting component structure:
+    /// //  D <- E
+    ///
+    /// assert!(!world.entity(entity).contains::<A>());  // A is removed as it requires B
+    /// assert!(!world.entity(entity).contains::<B>());
+    /// assert!(!world.entity(entity).contains::<C>());  // C is removed as it's required by B
+    /// assert!(world.entity(entity).contains::<D>());   // D is kept as E still requires it
+    /// assert!(world.entity(entity).contains::<E>());
+    /// ```
+    ///
+    /// This method is useful for removing a component along with its entire requirement tree and any
+    /// components that depend on them, while preserving components that are still required by parts of
+    /// the entity outside the removal tree.
+    pub fn require_descendant_remove<T: Bundle>(&mut self) -> &mut Self {
```

Comment:
I'm not particularly fond of this name, as it clashes with the concept of descendant in `bevy_hierarchy`.

Reply:
Yes. I just checked and indeed descendal only destroys descendants without destroying the entity itself. I renamed the function that cleans up and up the require tree to avoid confusion